f8a_report/sentry_report_helper.py

```diff
@@ -0,0 +1,141 @@
+                errors['id'] = item['id']
+                errors['last_seen'] = item['lastSeen']
+                errors[events['pods_impacted']] = item['metadata']['type'] + ": " + \
+                    item['metadata']['value'] if item['metadata'].get('type')\
+                    else item['metadata']['title']
+                errors['stacktrace'] = events['stacktrace']
+                # Detecting the endpoint services
+                server_name = "-".join(events['pods_impacted'].split("-")[:-2])
+                result['error_report'][server_name] = result['error_report'][server_name] \
+                    if result['error_report'].get(server_name) else {}
+                # Calculating total errors
+                result['error_report'][server_name]['total_errors'] = \
+                    result.get('error_report').get(server_name).get('total_errors', 0) + 1
+                if not result['error_report'][server_name].get('errors'):
+                    result['error_report'][server_name]['errors'] = []
+                result['error_report'][server_name]['errors'].append(errors)
+        except KeyError as e:
+            logger.error('Key not found while parsing. Reason: %r' % e)
+            # Saving the final report in the relevant S3 bucket
+
+        try:
+            obj_key = '{depl_prefix}/{type}/{report_name}.json'.format(
+                depl_prefix=self.s3.deployment_prefix, type=report_type, report_name=report_name
+            )
+            self.s3.store_json_content(content=result, obj_key=obj_key,
+                                       bucket_name=self.s3.report_bucket_name)
+        except Exception as e:
+            logger.exception('Unable to store the report on S3. Reason: %r' % e)
+
+        return result
+
+    def retrieve_events(self, issue_id):
+        """Retrieve results for issue events."""
+        events = {'stacktrace': ''}
+
+        try:
+            # Invoke Sentry API to run the event collection
+            auth = 'Bearer {token}'.format(token=self.sentry_token)
+            resp = requests.get(url=self.sentry_api_tags + issue_id + '/events/latest/',
+                                headers={"Authorization": auth})
+            # Check for status code
+            # If status is not success, log it as an error
+            if resp.status_code == 200:
+                logger.info('Successfully invoked Sentry API \n {resp}'.format(resp=resp.json()))
+                output = resp.json()
+            else:
+                logger.error('Error received from Sentry API \n {resp}'.format(resp=resp.json()))
+        except requests.exceptions.RequestException as e:
+            logger.error('Unable to invoke Sentry API3. Reason: %r' % e)
+        except requests.exceptions.Timeout as e:
+            logger.error('Timeout occured while invoking Sentry API. Reason: %r' % e)
+
+        try:
+            # retrieving server name info
+            for item in output['tags']:
```

Comment:
@animuk output['tags'] can cause a NameError if there is an exception handled before line no 106. Please handle this.

Reply:
handled.